sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClientTest.java

```diff
@@ -68,36 +66,37 @@ public void listRoleDefinitions(HttpClient httpClient) {
     @ParameterizedTest(name = DISPLAY_NAME)
     @MethodSource("com.azure.security.keyvault.administration.KeyVaultAdministrationClientTestBase#createHttpClients")
     public void setRoleDefinition(HttpClient httpClient) {
-        if (getTestMode() != TestMode.PLAYBACK) {
-            // Currently there is no test cloud environment for Managed HSM.
-            // TODO: Remove once there is a proper cloud environment available.
-            return;
-        }
+        Assumptions.assumeTrue(isHsmDeployed);
 
         asyncClient = getClientBuilder(httpClient, false).buildAsyncClient();
+        String roleDefinitionName = (getTestMode() == TestMode.PLAYBACK || getTestMode() == TestMode.RECORD)
+            ? "41424788-d6f0-4218-95b9-b597a3697fe0"
+            : UUID.randomUUID().toString();
 
-        String roleDefinitionName = "91d62511-feb2-456f-80a0-5b17bbaa50ec";
-
-        // Create a role definition.
-        asyncClient.setRoleDefinition(KeyVaultRoleScope.GLOBAL, roleDefinitionName)
-            .subscribe(roleDefinition -> {
-                assertNotNull(roleDefinition);
-                assertNotNull(roleDefinition.getId());
-                assertEquals(roleDefinitionName, roleDefinition.getName());
-                assertEquals(KeyVaultRoleDefinitionType.MICROSOFT_AUTHORIZATION_ROLE_DEFINITIONS,
-                    roleDefinition.getType());
-                assertTrue(roleDefinition.getAssignableScopes().contains(KeyVaultRoleScope.GLOBAL));
-                assertEquals(KeyVaultRoleType.CUSTOM_ROLE, roleDefinition.getRoleType());
-                assertEquals(roleDefinitionName, roleDefinition.getRoleName());
-
-                // Clean up the role definition.
+        try {
+            // Create a role definition.
+            StepVerifier.create(asyncClient.setRoleDefinition(KeyVaultRoleScope.GLOBAL, roleDefinitionName))
+                .assertNext(roleDefinition -> {
+                    assertNotNull(roleDefinition);
```

Comment:
nit: This isn't needed. `assertNext` won't trigger if the next emitted element in the reactive stream is null, in that case this would fail the test before entering the lambda.